### PR TITLE
Template REST task _ BNJC-23

### PR DESCRIPTION
**Pull request by ML6503**
refactoring of template for TS

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  "type": "commonjs",
+  "description": "REST service starter for RS School",
+  "main": "src/server.ts",
+  "scripts": {
+    "start": "ts-node src/server.ts",
+    "lint": "eslint --ext .js,.ts --fix src/",
+    "test": "cross-env DEBUG=rs:* jest --testMatch \"<rootDir>/test/e2e/test/*.test.js\" --noStackTrace --runInBand",
+    "test:auth": "cross-env DEBUG=rs:* TEST_MODE=auth jest --noStackTrace",
+    "build": "tsc -p .",
+    "dev": "npx nodemon src/server.ts"
```
Comment (AlreadyBored) [inline package.json:17]: Are you sure we need npx here?

File: src/app.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import * as express from 'express';
+import * as swaggerUI from 'swagger-ui-express';
+import * as path from 'path';
+import * as YAML from 'yamljs';
+import userRouter from './resources/users/user.router';
+
+const app: express.Express = express();
+const swaggerDocument = YAML.load(path.join(__dirname, '../doc/api.yaml'));
+// const swaggerDocument = YAML.load('../doc/api.yaml');
```
Comment (AlreadyBored) [inline src/app.ts:9]: Commented code?

File: src/app.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+app.use('/', async (req, res, next) => {
+  if (req.originalUrl === '/') {
+    res.send('Service is running!');
+    return;
+  }
+  next();
+});
+
+app.use('/', userRouter);
+// registerUserRouter(app);
```
Comment (AlreadyBored) [inline src/app.ts:23]: Commented code?

File: src/resources/users/user.router.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// const router = require('express').Router();
```
Comment (AlreadyBored) [inline src/resources/users/user.router.ts:1]: Commented code?

File: src/resources/users/user.interface.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+export interface IUser {
+  id: string;
+  name: string;
+  login: string;
+}
+
+export interface IUserDB extends IUser {
+  email: string;
+}
```
Comment (AlreadyBored) [inline src/resources/users/user.interface.ts:9]: At the moment we don't need to begin interfaces names with "I". And why do we need to have 2 interfaces here?

File: src/resources/users/user.model.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { v4 as uuid } from 'uuid';
+import { IUser } from './user.interface';
+
+export default class User {
+  private readonly id: string;
+
+
+
```
Comment (AlreadyBored) [inline src/resources/users/user.model.ts:8]: We don't need so many empty lines in a row :)

File: src/resources/users/user.model.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  private readonly name: string;
+
+
```
Comment (AlreadyBored) [inline src/resources/users/user.model.ts:11]: Same as above

File: src/resources/users/user.model.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  private readonly login: string;
+
```
Comment (AlreadyBored) [inline src/resources/users/user.model.ts:13]: Same as above

File: src/server.ts [hunk 9]
```diff
@@ -0,0 +1,8 @@
+import { config } from './common/config';
+import app from './app';
+
+const {PORT} = config;
```
Comment (AlreadyBored) [inline src/server.ts:4]:
```suggestion
const { PORT } = config;
```

File: src/app.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import * as express from 'express';
+import * as swaggerUI from 'swagger-ui-express';
+import * as path from 'path';
+import * as YAML from 'yamljs';
```
Comment (AlreadyBored) [inline src/app.ts:4]: Consider not to use syntax like `import * as something` for default import

File: tsconfig.json [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    "target": "ES2020",
+    "lib": ["ES2020"],
+    "moduleResolution": "node",
+    "sourceMap": true,
+    "outDir": "./dist",
+    "baseUrl": "./src",
+    "incremental": true
+  },
+  "include": ["src/**/*.ts"],
+  "exclude": ["node_modules"]
```
Comment (AlreadyBored) [inline tsconfig.json:18]: `exclude` only excludes files that are included by `include` field, please, remove this line

**Review by AlreadyBored — CHANGES_REQUESTED** [review]
Please, look at the comments + remove content of `test` folder, those tests are outdated

File: .env [hunk 12]
```diff
@@ -0,0 +1,6 @@
+NODE_ENV=development
+PORT=4000
+AUTH_MODE=false
+JWT_SECRET_KEY=secret-key
+# Set your database connection information here
+MONGO_CONNECTION_STRING=your-mongo-db-connection-string
```
[the remark this answers is not in the document]
Comment (ML6503) [reply .env:6]: removed

File: .github/workflows/ci.yml [hunk 13]
```diff
@@ -0,0 +1,43 @@
+name: CI
```
[the remark this answers is not in the document]
Comment (ML6503) [reply .github/workflows/ci.yml:1]: removed

File: .github/workflows/md5.yml [hunk 14]
```diff
@@ -0,0 +1,23 @@
+name: MD5
```
[the remark this answers is not in the document]
Comment (ML6503) [reply .github/workflows/md5.yml:1]: removed

File: .gitignore [hunk 15]
```diff
@@ -0,0 +1 @@
+node_modules
```
[the remark this answers is not in the document]
Comment (ML6503) [reply .gitignore:1]: done

File: .vscode/settings.json [hunk 16]
```diff
@@ -0,0 +1,9 @@
+{
```
[the remark this answers is not in the document]
Comment (ML6503) [reply .vscode/settings.json:1]: removed

File: package.json (hunk 1 above)
Comment (ML6503) [reply package.json:17]: removed)

File: src/app.ts (hunk 2 above)
Comment (ML6503) [reply src/app.ts:9]: sorry.. shame on me.. it is removed

File: src/app.ts (hunk 3 above)
Comment (ML6503) [reply src/app.ts:23]: removed

File: src/resources/users/user.router.ts (hunk 4 above)
Comment (ML6503) [reply src/resources/users/user.router.ts:1]: sorry.. removed

File: src/resources/users/user.interface.ts (hunk 5 above)
Comment (ML6503) [reply src/resources/users/user.interface.ts:9]: Renamed as requested. Used both in user.router and user.model

File: src/resources/users/user.model.ts (hunk 6 above)
Comment (ML6503) [reply src/resources/users/user.model.ts:8]: sorry. removed

File: src/resources/users/user.model.ts (hunk 7 above)
Comment (ML6503) [reply src/resources/users/user.model.ts:11]: removed

File: src/resources/users/user.model.ts (hunk 8 above)
Comment (ML6503) [reply src/resources/users/user.model.ts:13]: removed

File: src/server.ts (hunk 9 above)
Comment (ML6503) [reply src/server.ts:4]: amended

File: src/app.ts (hunk 10 above)
Comment (ML6503) [reply src/app.ts:4]: Noted. Amended

File: tsconfig.json (hunk 11 above)
Comment (ML6503) [reply tsconfig.json:18]: removed

**Pull request merged**